jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -947,17 +946,16 @@ def _fit_model_to_cube(self, add_data):
                 temp_label = "{} ({}, {})".format(self.results_label, m["x"], m["y"])
                 self.app.fitted_models[temp_label] = m["model"]
 
-        count = max(map(lambda s: int(next(iter(re.findall(r"\d$", s)), 0)),
-                        self.data_collection.labels)) + 1
-
-        label = self.app.return_data_label(f"{self.results_label}[Cube]", ext=count)
-
-        # Create new glue data object
-        output_cube = Data(label=label,
-                           coords=fitted_spectrum.wcs,
-                           flux=fitted_spectrum.flux.value)
-        output_cube.get_component('flux').units = fitted_spectrum.flux.unit.to_string()
+        w = fitted_spectrum.wcs
+        if w.pixel_n_dim == 1:
+            # TODO: Is this really correct?
+            # Need to flip X, Y to be consistent with input cube.
+            outflux = np.moveaxis(fitted_spectrum.flux, 1, 0)
```

Comment:
I think this is what @camipacifici wanted but when I try to build a regression test case for this, the Data object added into data_collection has a shape that does not match the input data cube. 🤯

Reply:
I think Cami's transposing problem is unrelated to the model fitting problem I am trying to fix here, so I am going to revert this.